Mathlib/NumberTheory/Harmonic/ZetaAsymp.lean

```diff
@@ -0,0 +1,336 @@
+  refine div_nonneg ?_ (rpow_nonneg ?_ _)
+  all_goals linarith [hx.1]
+
+lemma term_welldef {n : ℕ} (hn : 0 < n) {s : ℝ} (hs : 0 < s) :
+    IntervalIntegrable (fun x : ℝ ↦ (x - n) / x ^ (s + 1)) volume n (n + 1) := by
+  rw [intervalIntegrable_iff_integrableOn_Icc_of_le (by linarith)]
+  refine (ContinuousAt.continuousOn fun x hx ↦ ContinuousAt.div ?_ ?_ ?_).integrableOn_Icc
+  · fun_prop
+  · apply continuousAt_id.rpow_const (Or.inr <| by linarith)
+  · exact (rpow_pos_of_pos ((Nat.cast_pos.mpr hn).trans_le hx.1) _).ne'
+
+section s_eq_one
+
+/-!
+## Evaluation of the sum for `s = 1`
+-/
+
+lemma term_one {n : ℕ} (hn : 0 < n) :
+    term n 1 = (log (n + 1) - log n) -  1 / (n + 1) := by
+  have hv : ∀ x ∈ uIcc (n : ℝ) (n + 1), 0 < x := by
+    intro x hx
+    rw [uIcc_of_le (by simp only [le_add_iff_nonneg_right, zero_le_one])] at hx
+    exact (Nat.cast_pos.mpr hn).trans_le hx.1
+  calc
+    _ = ∫ x : ℝ in n..(n + 1), (x - n) / x ^ 2 := by
+      simp_rw [term, one_add_one_eq_two, ← Nat.cast_two (R := ℝ), rpow_nat_cast]
+    _ = ∫ x : ℝ in n..(n + 1), (1 / x - n / x ^ 2) := by
+      refine intervalIntegral.integral_congr (fun x hx ↦ ?_)
+      field_simp [(hv x hx).ne']
+      ring
+    _ = (∫ x : ℝ in n..(n + 1), 1 / x) - n * ∫ x : ℝ in n..(n + 1), 1 / x ^ 2 := by
+      simp_rw [← mul_one_div (n : ℝ)]
+      rw [intervalIntegral.integral_sub]
+      · simp_rw [intervalIntegral.integral_const_mul]
+      · exact intervalIntegral.intervalIntegrable_one_div (fun x hx ↦ (hv x hx).ne') (by fun_prop)
+      · exact (intervalIntegral.intervalIntegrable_one_div
+          (fun x hx ↦ (sq_pos_of_pos (hv x hx)).ne') (by fun_prop)).const_mul _
+    _ = (log (↑n + 1) - log ↑n) - n * ∫ x : ℝ in n..(n + 1), 1 / x ^ 2 := by
+      congr 1
+      rw [integral_one_div_of_pos, log_div]
+      all_goals positivity
+    _ = (log (↑n + 1) - log ↑n) - n * ∫ x : ℝ in n..(n + 1), x ^ (-2 : ℝ) := by
+      congr 2
+      refine intervalIntegral.integral_congr (fun x hx ↦ ?_)
+      rw [rpow_neg, one_div, ← Nat.cast_two (R := ℝ), rpow_nat_cast]
+      exact (hv x hx).le
+    _ = log (↑n + 1) - log ↑n - n * (1 / n - 1 / (n + 1)) := by
+      rw [integral_rpow]
+      · simp_rw [sub_div, (by norm_num : (-2 : ℝ) + 1 = -1), div_neg, div_one, neg_sub_neg,
+          rpow_neg_one, ← one_div]
+      · refine Or.inr ⟨by norm_num, not_mem_uIcc_of_lt ?_ ?_⟩
+        all_goals positivity
+    _ = log (↑n + 1) - log ↑n - 1 / (↑n + 1) := by
+      congr 1
+      field_simp
+
+lemma term_sum_one (N : ℕ) : term_sum 1 N = log (N + 1) - harmonic (N + 1) + 1 := by
+  induction' N with N hN
+  · simp_rw [term_sum, Finset.sum_range_zero, harmonic_succ, harmonic_zero, Nat.zero_eq,
+      Nat.cast_zero, zero_add, Nat.cast_one, inv_one, Rat.cast_one, log_one, sub_add_cancel]
+  · unfold term_sum at hN ⊢
+    rw [Finset.sum_range_succ, hN, harmonic_succ (N + 1),
+      term_one (by positivity : 0 < N + 1)]
+    push_cast
+    ring_nf
+
```

Comment:
My personal preference is to add docstrings to the "main" result of any section. (Feel free to ignore this suggestion.)